### DISCUSSION TITLE
命令标签分享

### DISCUSSION THREAD

**Discussion by MuiseDestiny** [MuiseDestiny/zotero-gpt#3] [Ideas]
如果你具有一定的Zotero开发经验，并制作了一个命令标签，可以PR到[tags](https://github.com/MuiseDestiny/zotero-gpt/tree/bootstrap/tags)下。

并在这里分享截图。

**Comment by MuiseDestiny** [comment]
#摘要提问[pos=0]

下面是一篇论文的摘要：
```js
const ztoolkit = Zotero.ZoteroGPT.data.ztoolkit
let getSelection = () => {
    return ztoolkit.Reader.getSelectedText(
 Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)
    );
}
console.log(getSelection())
getSelection()

```

---

请问它的主要工作是什么，在什么地区，时间范围是什么，使用的数据是什么，创新点在哪？

请你用下列示例格式回答我：
主要工作：反演AOD；
地区：四川盆地；
时间：2017~2021；
数据：Sentinel-2卫星数据；
创新：考虑了BRDF效应。



![image](https://user-images.githubusercontent.com/51939531/228212379-166a6f56-2627-4c2a-a263-c0da14af64e9.png)

**Reply by chikun0** [reply]
大佬，请问我们普通人怎么把这些标签贴到我们的zotero gpt栏目里呀

**Reply by MuiseDestiny** [reply]
会内置标签的。先用内置，等稳定了再学习怎么建立标签或者导入别人的

**Reply by zzzSEA** [reply]
为什么我运行，是解释这段代码啊？

**Reply by yingjiecs** [reply]
必须打开pdf选择摘要才行。能不能使用条目的abstract而不是pdf选择的abstract

**Reply by dudu0524** [reply]
大佬，怎么删除标签呀？不小心弄了很多错的进去

**Reply by Karllihaiq** [reply]
右键条目是删除

**Reply by Deepcalcite** [reply]
请问有这段表示摘要的代码的解释吗？我使用后表示无法读取该代码，顺带想问一下在zotero的命令里面，有没有获取全文的代码呢？我在很多地方都没有找到

**Comment by MuiseDestiny** [comment]
#摘要转综述[pos=1]

下面是一篇论文的摘要：
```js
const ztoolkit = Zotero.ZoteroGPT.data.ztoolkit
let getSelection = () => {
    return ztoolkit.Reader.getSelectedText(
 Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)
    );
}
console.log(getSelection())
getSelection()

```

---

现在我想引用它写一篇文献综述，请你帮我写几句话总结这篇论文的工作，作者没有告诉你，你可以用XX代替。我要中文的。要求50字左右。

![image](https://user-images.githubusercontent.com/51939531/228208668-82a36871-fd49-4ae5-b05c-07bb1c01f706.png)

**Reply by Cocokou0510** [reply]
大大您好，我用了这个代码，且选中了摘要文字，但是GPT瞎编了无关的内容返回给我，请问这个怎么解决qwq

**Reply by p115760** [reply]
![image](https://github.com/MuiseDestiny/zotero-gpt/assets/161483762/5346a89d-22eb-4494-a09d-c91a8ffe361f)
<img width="497" alt="1" src="https://github.com/MuiseDestiny/zotero-gpt/assets/161483762/661536b1-b045-4c50-bfa6-badfc773bc7f">

**Reply by p115760** [reply]
操作执行提示这个，没有成功

**Reply by HuangHuahang** [reply]
这篇论文研究了使用 Zotero 工具包中的 ztoolkit.Reader.getSelectedText 方法来获取阅读器中选定的文本。该方法对文献管理和学术研究领域具有重要意义，为 XX 领域的学者提供了新的工具和资源。

**Comment by MuiseDestiny** [comment]
#期刊评价
请问下面这个期刊什么水平：

```js
ZoteroPane.getSelectedItems()[0].getField("publicationTitle")
```

请给出它的影响因子和你对它的评价。


![image](https://user-images.githubusercontent.com/51939531/228260698-13e18128-c885-4a4a-9bea-2f7e4a06bb2b.png)

**Reply by Zyyuan515** [reply]
很抱歉，作为一个文本AI助手，我无法实时获取特定期刊的影响因子信息。您可以通过访问类似Web of Science、PubMed等学术数据库或使用期刊评价工具来获取特定期刊的影响因子信息。根据期刊的影响因子、被引频次、编辑委员会以及刊登的论文质量等因素综合评价期刊的水平。如果您对期刊具体的质量和声誉有更多要求，您可以考虑查找该期刊的评价和反馈。
为什么我的回答是这样的呢

**Reply by jackhanyuan** [reply]
您好，zotero7.0正式版中，最新的GPT，版本1.2.2，创建的标签会报错，运行不了，js代码块的代码不会被执行。

**Reply by Invincible-Dhh** [reply]
请问返回的web链接无法打开这个正常吗

**Comment by panwanke** [comment]
#总结实验方法[pos=0]
请用子弹列表的形式提取该段落的信息，包括，被试数量，实验设计(如, 2(Gender: male vs female) X 2(Matching Condition: matching or mismatching))，实验流程(如，被试首先观看刺激，然后进行反应，最后呈现反馈给被试)。务必使用中文回答。
下面是论文的方法部分：
```js
const ztoolkit = Zotero.ZoteroGPT.data.ztoolkit
let getSelection = () => {
    return ztoolkit.Reader.getSelectedText(
 Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)
    );
}
console.log(getSelection())
getSelection()

```

**Reply by Cocokou0510** [reply]
大佬您好，你是做心理学或行为学实验的吗？请问您使用什么代码分析某篇文章的摘要或者几篇文章的内容？我尝试让gpt帮忙总结自变量，中介，调节之类的，但是它好像听不懂，也不能识别

**Reply by panwanke** [reply]
要实现这个的话在 zotero里感觉不是很方便，可以学习一下 langchain。

**Comment by MuiseDestiny** [comment]
用途：推荐添加嵌套标签
名称：AddTags
内容：[这里](https://github.com/MuiseDestiny/zotero-gpt/blob/bootstrap/labels/AddTags.txt)


![image](https://user-images.githubusercontent.com/51939531/230590806-33e4704b-e102-4362-834b-4fa8f6acd8a7.png)

**Reply by systemoutprintlnhelloworld** [reply]
代碼鏈接變更為https://github.com/MuiseDestiny/zotero-gpt/blob/bootstrap/tags/AddTags.txt

**Reply by broke-why** [reply]
新更新的版本给出的嵌套标签似乎不能直接自动执行，请问将代码复制下来后，在哪里可以执行实现标签添加呢？

**Reply by l7ee** [reply]
多组嵌套标签的话有排序功能吗

**Reply by Hsuxu** [reply]
```js
Zotero.ZoteroGPT.views.messages = [];
JSON.stringify(Zotero.ZoteroStyle.data.views.tagsUI.plainTags.slice(0, 20))
```
不能正确执行，获取不到当前的tag, 错误如下。

![Snipaste_2024-08-30_10-50-03](https://github.com/user-attachments/assets/3bd3fc6c-fecb-43a8-a75f-5a26616b5693)
- zotero 版本：`7.0.3 (64-bit)`

**Comment by MaliquyChen** [comment]
#出題
以下是一段敘述性的文字
```js
const ztoolkit = Zotero.ZoteroGPT.data.ztoolkit
let getSelection = () => {
    return ztoolkit.Reader.getSelectedText(
 Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)
    );
}
console.log(getSelection())
getSelection()
```
請根據上述文字出一個包含A,B,C,D四個選項的選擇題，然後給出正確答案以及詳解
reply in 繁體中文

![image](https://user-images.githubusercontent.com/92513632/231698979-d0ea9c80-608f-47af-b667-4a04a40dcfc2.png)

**Reply by WengMixin** [reply]
#Summeraize[position=0][color=#009FBD][trigger=]
You are an helpful assistant who is good at summarizing literature. Context information is below.
${
Meet.Global.views.messages = [];
Meet.Zotero.getRelatedText("Find the main parts of the full text that can best help me summarize.")
}
Write a comprehensive summary response to a given query, using the contextual information provided. Make sure using the following titles as your summary's output but not limited to the content and language, they are: KeyWords,Prerequisites,Summary,Related reading. Each titles have the following minimum requirements:

- KeyWords: List some keywords that can represent the whole essay.
- Prerequisites: List the necessary prerequisites in order to be able to fully understand the entire content.
- Summary: Summarize the whole provided Context elegantly and comprehensively.
- Related reading: Search some related reading sources on your data base with their title and link.

Reply in zh-CN
---
![test](https://user-images.githubusercontent.com/119753827/235205087-3d31b616-d47c-4725-b180-ca2e3e822e96.png)

**Reply by 006Evan** [reply]
太棒了吧

**Reply by Fan-tas-tic** [reply]
请问这是怎么使用的呢？直接再打开的PDF里面点击Summary吗？我设置了同样的代码点击却没有反应！

**Comment by Gaozhongpai** [comment]
如何实现打开一个PDF后，ChatGPT 自动读入整篇论文，然后输入问题后直接可以交互，而不用点击标签？

**Reply by MuiseDestiny** [reply]
有触发词，输入  本文xxxx 回车即可

**Reply by Gaozhongpai** [reply]
多谢，可以是英文吗？ This paper

**Reply by Gaozhongpai** [reply]
请问哪里有什么文档解释以下函数？
```

Meet.Global.views.messages = [];
Meet.Zotero.getRelatedText()
```

```
let getSelection = () => {
    return ztoolkit.Reader.getSelectedText(
 Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)
    );
}
```

```
window.gptInputString = Zotero.ZoteroGPT.views.inputContainer.querySelector("input").value
Zotero.ZoteroGPT.views.messages = [];

Zotero.ZoteroGPT.utils.getRelatedText(
window.gptInputString 
)
```

请问什么函数可以获取全文，而不是选中的文字？

非常感谢！

**Reply by MuiseDestiny** [reply]
> 多谢，可以是英文吗？ This paper

可以的，长按askpdf修改一下trigger

**Reply by MuiseDestiny** [reply]
https://github.com/MuiseDestiny/zotero-gpt/blob/bootstrap/src/modules/Meet/api.ts

这里有部分函数解释

**Reply by QingYu-sunshine** [reply]
您好，请问您实现您的想法了吗？可以分享一下实现的代码吗？

**Comment by river7816** [comment]
一个数学公式Solver，能解决latex输出问题
```
#Math Solver[position=9][color=#0C669B][trigger=]
Read these content:
${
Meet.Zotero.getPDFSelection() ||
Meet.Global.views.messages[0].content
}
---
From now on, if you need to write a mathematical expression, use latex notation and follow these rules:
1. If it is a block equation, display it in a single P element and wrap it with double dollar signs like this:
$$e=mc^{2}$$

2. If it is an inline equation, wrap it with a double dollar sign, like this: $e=mc^{2}$

You should first explain the proof or equations step by step and demonstrations of various techniques to better understand or apply these concepts/equations.
```

**Reply by zhanghay** [reply]
太好用了！

**Comment by jhs1965net** [comment]
Hi MuiseDestiny , really appreciate your work , it save me a big amount of time and one my research.
Sorry but I can't open get with the hot keys you give , / is in number 7 and ? is in' , Spanish keyboard.
Thanks a lot for your great work

James

**Reply by MuiseDestiny** [reply]
Hi. I released 0.2.9.

![image](https://github.com/MuiseDestiny/zotero-gpt/assets/51939531/21f11f7a-29a4-46ba-ae89-0242847a9e94)

**Reply by jhs1965net** [reply]
Sorry to bother but I have this answer

<strong xmlns="http://www.w3.org/1999/xhtml">insufficient_quota</strong><br xmlns="http://www.w3.org/1999/xhtml" />
You exceeded your current quota, please check your plan and billing details.

**Reply by MuiseDestiny** [reply]
https://platform.openai.com/account/usage

see here
![image](https://github.com/MuiseDestiny/zotero-gpt/assets/51939531/368d2a62-8515-41dc-971d-953a3e791913)

**Reply by jhs1965net** [reply]
Justo to note that annotations answers are in Chinese , same as translation, and items and selection

**Reply by MuiseDestiny** [reply]
Long press a tag to edit its inner prompt. You can read readme to understand it better.

**Reply by jhs1965net** [reply]
Hi MuiseDestiny
I was using the plugin and it is awesome but it seems that it works better in windows , as a user of a Mac is it possible to check it in Mac because there are many functionalities that doesn't work.
Does the plugin works better in better notes?
thanks a lot.
James

**Comment by jhs1965net** [comment]
You are great , thanks RegardsJamesEnviado desde mi iPhoneEl 16 may 2023, a las 4:10, Polygon ***@***.***> escribió:﻿
Hi. I released 0.2.9.

**Reply by jhs1965net** [reply]
resolved


You are a genius

**Comment by jhs1965net** [comment]
Done , thank you , really appreciateEnviado desde mi iPhoneEl 16 may 2023, a las 11:28, Polygon ***@***.***> escribió:﻿
https://platform.openai.com/account/usage
see here

**Comment by jhs1965net** [comment]
:)
Is it possible to have the password for connected papers?
***@***.*** ***@***.***>

Kindly

**Comment by Ltitor** [comment]
用途:针对选中的段落在全文的范畴中分析该段落对于全文的作用和在全文中的意义。有助于在不脱离上下文的前提下对特定段落进行询问而不至于让gpt把范畴集中在全文。
名称：AskSelection


```
#🪐AskSelection[position=0][color=#009FBD][trigger=]
You are a helpful assistant. Context information is below.

window.gptInputString = Zotero.ZoteroGPT.views.inputContainer.querySelector("input").value
Zotero.ZoteroGPT.views.messages = [];

Zotero.ZoteroGPT.utils.getRelatedText(
window.gptInputString 
)
Current date:   
String(new Date())
The above is the whole article, here a selected paragraph in the article above:
Meet.Zotero.getPDFSelection();

Make sure to cite results using [number] notation after the reference. If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information. 

Answer the question regarding to the selected paragraph:
window.gptInputString 
Reply in english
```

**Comment by mbs0221** [comment]
```
#会议介绍[position=102][color=#ffcccc][trigger=]

介绍一下下面这个会议：  

ZoteroPane.getSelectedItems()[0].getField("proceedingsTitle")

请列出你所知道的相关领域的国际会议和期刊。

请列出你所知道的相关领域的主要研究人员。

请列出你所知道的相关领域的知名实验室。

---

应用场景: 选中主界面条目时候
```

**Comment by jhs1965net** [comment]
Hi there , thanks for all the knowledge
I have upgraded to zotero 7 to check how it works , plenty of problems , specially with the cite keys and 

The button in zotero 7 , last OS X is not triggering the plug in

Thanks a lot 

James Hamilton

**Comment by jaysonfraser** [comment]
按照上面的教程，自己用了下，发现不对。
![zongshu](https://github.com/MuiseDestiny/zotero-gpt/assets/64569102/4515bbfc-164a-4911-940e-325c9c16f3a5)

**Comment by jaysonfraser** [comment]
![zhaiyaotiwen](https://github.com/MuiseDestiny/zotero-gpt/assets/64569102/104ed139-6c29-4cae-939a-f8604e96396f)

**Comment by jaysonfraser** [comment]
![222](https://github.com/MuiseDestiny/zotero-gpt/assets/64569102/423d4acb-afe3-4c7e-a0bc-160f4d910c1d)

**Reply by sxpmwmh** [reply]
你的用法不对，应该把摘要复制到上面的空白框里面再点“摘要提问”

**Comment by jaysonfraser** [comment]
![333](https://github.com/MuiseDestiny/zotero-gpt/assets/64569102/3b3bebe8-902c-4dbd-be82-b0a505e62fcf)
用默认的还是不对，请问这主要是什么原因

**Comment by MuiseDestiny** [comment]
```
@Zotero GPT Tag by Polygon
#文章的创新点与不足[position=0][color=#0EA293][trigger=/^(本文|这篇文章|论文)/]
You are a helpful assistant. Context information is below.

<code>Meet.Zotero.getRelatedText("请总结本文有什么创新以及不足")</code>

Using the provided context information, write a comprehensive reply to the given query. Make sure to cite results using [number] notation after the reference. If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information.

Answer the question: 请总结本文有什么创新以及不足

Reply in zh-CN
```

```
@Zotero GPT Tag by Polygon
#总结热解机理[position=0][color=#0EA293][trigger=/^(本文|这篇文章|论文)/]
You are a helpful assistant. Context information is below.

<code>Meet.Zotero.getRelatedText("热解机理")</code>

Using the provided context information, write a comprehensive reply to the given query. Make sure to cite results using [number] notation after the reference. If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information.

Answer the question: 请总结热解机理

Reply in zh-CN
```

```
@Zotero GPT Tag by Polygon
#总结热解动力学模型[position=0][color=#0EA293][trigger=/^(本文|这篇文章|论文)/]
You are a helpful assistant. Context information is below.

<code>Meet.Zotero.getRelatedText("热解动力学模型")</code>

Using the provided context information, write a comprehensive reply to the given query. Make sure to cite results using [number] notation after the reference. If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information.

Answer the question: 请总结热解动力学模型

Reply in zh-CN
```

**Reply by Rosenls** [reply]
为什么我的gpt不会读取我选定的论文？不管怎么换2代码都没用

**Reply by Rosenls** [reply]
就一直这样：很抱歉，由于上下文中没有提供与文章有关的详细信息，我无法回答关于文章内容的问题。如果您可以提供更多细节或具体的问题，我将尽力为您提供帮助。

**Reply by zfb132** [reply]
@Rosenls 重新开个帖子详细说明你的信息。例如软件和插件版本号，你在哪里调出的gpt面板，输入的什么内容，点击的哪个标签（有没有修改过内置标签），右下角的弹窗显示什么。

**Comment by Komorebi-yaodong** [comment]
请问一下大佬们能帮我看一下我写的这一段有什么问题吗？

**Reply by Komorebi-yaodong** [reply]
#🤖论文提炼[position=10][color=#F13433][trigger=/^(本文|这篇文章|论文)/]
You are a helpful assistant. Context information is below.
${
Meet.Zotero.getRelatedText(Meet.Global.input)
}
Please answer the following questions in the form of a table: 1. What is the problem that the article aims to solve? 2. What is the research background and current status related to the target problem? What are the drawbacks of existing solutions? What is the main starting point and idea of the article? 5. What is the novelty of the paper work reflected in? 6. Is the core method adopted by the author to solve the target problem, and how to conduct testing and verification persuasive?Make sure to cite results using [number] notation after the reference. 

Reply in zh-CN

**Reply by zfb132** [reply]
```text
#🤖PaperSummary[position=20][color=#F13433][trigger=/^(本文|这篇文章|论文)/]
<code>
Meet.Zotero.getRelatedText(Meet.Global.input) 
</code>

Please answer the following questions in the form of a table: 
1. What is the problem that the article aims to solve? 
2. What is the research background and current status related to the target problem? 
3. What are the drawbacks of existing solutions? 
4. What is the main starting point and idea of the article? 
5. What is the novelty of the paper work reflected in? 
6. Is the core method adopted by the author to solve the target problem, and how to conduct testing and verification persuasive?

Make sure to cite results using [number] notation after the reference.  

Reply in <code>Zotero.locale</code>
```
![image](https://github.com/MuiseDestiny/zotero-gpt/assets/18099238/39ba5f5e-02d1-4103-ae5f-f6fa4d01a1e2)

**Reply by hlovem666** [reply]
请问为什么会出现这个：抱歉，我不能访问外部数据库或执行代码。如果您能提供相关的信息和数据，我会非常乐意帮助您回答问题。

**Comment by brunoalbertopeixoto** [comment]
@MuiseDestiny I thought your app was wonderful. I'm Brazilian and I don't know Chinese, English or programming. I couldn't get it to work without having a ChatGPT account. I imagine it would be very difficult to program, but I have some suggestions. Is it possible to use other LLMAs and API models such as: https://www.llama-api.com/, https://blog.perplexity.ai/blog/introducing-pplx-api and other free opensouce. Another suggestion, as you master developing plugins for Zotero, you could develop a Video Annotation to Zotero application or resume development of this: http://dancohen.org/2008/03/18/vertov-brings-video-annotation-to-zotero/ . You can download the latest version from web.archive.org. I hope you have time to develop these updates. It will help many people.






> 如果你具有一定的Zotero开发经验，并制作了一个命令标签，可以PR到[tags](https://github.com/MuiseDestiny/zotero-gpt/tree/bootstrap/tags)下。
> 
> 并在这里分享截图。

**Comment by Edilson-R** [comment]
I need ASK PDF to return each exact section where the references were taken, including the DOCUMENT PAGE. how to do this?

**Comment by jackhanyuan** [comment]
总结收集了一些zotero-gpt命令标签，欢迎star：https://github.com/jackhanyuan/zotero-better-plugin?tab=readme-ov-file#command-tag

**Reply by Davidscat** [reply]
十分感谢

**Reply by yunsgao** [reply]
感谢老哥！

**Comment by huachuman** [comment]
Hi, I'm not able to get anything working

Here's an example tag I've tried:

>#📃Abxtract[color=#9F949E][trigger=]
>
>Please summarize Abstract with the template:
>
>Abstract: ${ Meet.Zotero.getItemField("abstractNote")}
>
>Aims:
>
>Findings:
>
>Method:

But the response I get: 
> It seems like the abstract text is not being displayed correctly. Please provide the actual content of the abstract, and I’ll help you fill out the form with concise information on the aims, findings, and method.

My tags from a long time ago that used to work are also not working:

e.g.

"#✍️SumFind[position=2][color=#E11299]
\```js
// Ensure that you have selected the summary section of the PDF.
Zotero.ZoteroGPT.utils.getPDFSelection()
\```


It just gives me the error
>Zotero.ZoterGPT.utils is undefined

**Comment by Invincible-Dhh** [comment]
请问 作者 AI 简记的 prompt是啥啊，感觉很棒

**Comment by whubao** [comment]
大佬，能不能导入pdf后自动让gpt总结论文创新点、研究动机、研究方法、研究结论、全文总结等，并直接在zotero的某些列显示。就像飞书加deepseek一样，自动完成论文总结到每列。

**Reply by MuiseDestiny** [reply]
试试Magic已经完全实现了你说的https://magiczotero.feishu.cn/wiki/WDx0wCN5Cizq6okjxUFcMThln3d?fromScene=spaceOverview

**Comment by amateurProfessor** [comment]
### Translation/Translate Command Tags -  English and German
Thank you for the great plugin!🙏🏼
```
#🇬🇧Translate[color=#F59FC0][trigger=/^translate/]
I want you to act as a scientific English-German translator. I will provide you with some paragraphs in one language and your task is to accurately and academically translate the paragraphs only into the other language. Do not repeat the original provided paragraphs after translation. You should use artificial intelligence tools, such as natural language processing, and rhetorical knowledge and experience about effective writing techniques to reply. I'll give you my paragraphs as follows, then you translate:

${
Meet.Global.input.replace("translate", "") ||
Meet.Zotero.getPDFSelection() ||
Meet.Global.views.messages[0].content
}$
```

```
#🇩🇪Übersetzen[color=#15DCD4][trigger=/^übersetze/]
Ich möchte, dass du als wissenschaftlicher Deutsch-Übersetzer fungierst. Ich werde dir einige Absätze in einer Sprache geben, und deine Aufgabe ist es, die Absätze akkurat und wissenschaftlich nur ins Deutsche zu übersetzen. Wiederhole die ursprünglichen Absätze nach der Übersetzung nicht. Du solltest Werkzeuge der künstlichen Intelligenz wie die natürliche Sprachverarbeitung und rhetorisches Wissen und Erfahrung über effektive Schreibtechniken verwenden, um zu antworten. Ich werde dir meine Absätze wie folgt geben, dann übersetze:

${
Meet.Global.input.replace("übersetze", "") ||
Meet.Zotero.getPDFSelection() ||
Meet.Global.views.messages[0].content
}$
```

**Comment by amateurProfessor** [comment]
### 📄Artikel fragen (Deutsch/German) ≙ "🪐AskPDF" - German/Deutsch

```
#📄ArtikelFragen[color=#297159][trigger=/^(fragArtikel|dieser artikel|diese arbeit|artikel fragen)/]
You are a helpful assistant. Context information is below.
${
Meet.Zotero.getRelatedText(Meet.Global.input || "Fasse diesen Artikel zusammen.")
}$
Using the provided context information, write a comprehensive reply to the given query. Make sure to cite results using [number] notation after the reference. If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information.

Answer the question: ${Meet.Global.input || "Fasse diesen Artikel zusammen."}$
Answer in German.
```

**Comment by amateurProfessor** [comment]
### 🇩🇪FragPDF (Full text) ≙ "🪐AskPDF (Full text)" - German/Deutsch
```
#🇩🇪FragPDF (Full text)[color=#F70EBD][trigger=/^(fragpdf|pdf fragen|volltext fragen)/]
You are a helpful assistant. Paper's full text from PDF is below.

${
(async () => {
    let pdfItems = []
    if (Zotero_Tabs.selectedType == "library") {
        const items = ZoteroPane.getSelectedItems()
        for (let item of items) {
            if (item.isPDFAttachment()) {
                pdfItems.push(item)
            } else if ((await item.getBestAttachmentState()).exists) {
                pdfItems.push(
                    await item.getBestAttachment()
                )
            }
        }
    } else if (Zotero_Tabs.selectedType == "reader"){
        pdfItems = [
            Zotero.Reader.getByTabID(Zotero_Tabs.selectedID)._item
        ]
    }
    let docs = []
    let res = ""
    for (let pdfItem of pdfItems) {
        const fullText = (await Zotero.PDFWorker.getFullText(pdfItem.id, null)).text.split("\n")
        const pageContent = (pdfItem.topLevelItem.firstCreator + " ("+ String(pdfItem.topLevelItem.getField("year")) + ") [" + String(pdfItems.indexOf(pdfItem) + 1) + "] "  + pdfItem.topLevelItem.getDisplayTitle() + "\n"+ fullText)
        res += pageContent + "\n\n"
        docs.push({
            pageContent: pageContent,
            metadata: {
                type: "id",
                id: pdfItem.topLevelItem.id
            }
        })
    }
    Meet.Global.views.insertAuxiliary(docs)
    return res
})()
}$

Using the provided context information, write a comprehensive reply to the given query.
Make sure to cite results using [number] notation after the reference. Please use the 'author(year)[number]' format when referring to the paper, such as 'Zhao (2025) [2]'.
Answer the question: ${Meet.Global.input || "Fasse den Volltext zusammen."}$.
Answer in German.
```

**Comment by amateurProfessor** [comment]
### 📋Clipboard, ✍️Selection, 📝Annotations, 📚Items - German-English Version
Uses both English and German triggers, if you want, you can change the name of the tags to English.
```
#📋Zwischenablage[color=#8F97D9][trigger=/^(zwischenablage|kopierter inhalt|clipboard|copied content)/]
Dies ist der Inhalt in meiner Zwischenablage:
${Meet.Zotero.getClipboardText()}$
---
${Meet.Global.input}$
```

```
#📝AusgewählteAnmerkungen[color=#52D9D5][trigger=/^(ausgewählte? anmerkungen|ausgewählte? markierungen|selected annotations|selected highlights)/]
These are PDF Annotation contents:
${
Meet.Zotero.getPDFAnnotations(Meet.Global.input.match(/(ausgewählte?|selected)/i))
}$

Please answer me in the language of my question. Make sure to cite results using [number] notation after the reference.
My question is: ${Meet.Global.input}$
```
```
#✍️Auswahl[color=#C209F2][trigger=/^(ausgewählter text|markierter text|selected text|highlighted text)/]
Read these content:
${
Meet.Zotero.getPDFSelection() ||
Meet.Global.views.messages[0].content
}$
---
Answer me in the language of my question. This is my question: ${Meet.Global.input}$
```
```
#📚Items[color=#826E56][trigger=/^(diese artikel|diese literatur|ausgewählte artikel|these items|selected items|this literature)/]
These are Zotero item informations:
${
(async function() {
    let items = ZoteroPane.getSelectedItems()
    const docs = Meet.Zotero.items2documents(items)
    Meet.Global.views.insertAuxiliary(docs)
    return docs.map(
        (doc, index) => "[" + String(index + 1) + "]" + " " +  doc.pageContent
    ).join("\n\n")
})()
}$
---

Please answer me using the lanaguage as same as my question. Make sure to cite results using [number] notation after the reference.
My question is: ${Meet.Global.input || "summary these papers"}$
```

**Comment by amateurProfessor** [comment]
### 📖Literature Review with English and German triggers, answers in English
```
#📖LiteratureReview[color=#45BDDC][trigger=/^(literaturübersicht für|review für|literature review of|review of)/]
These are Zotero item informations:
${
(async function() {
    let items = ZoteroPane.getSelectedItems()
    const docs = Meet.Zotero.items2documents(items)
    Meet.Global.views.insertAuxiliary(docs)
    return docs.map(
        (doc, index) => "[" + String(index + 1) + "]" + " " +  doc.pageContent
    ).join("\n\n")
})()
}$

---

Please read these documents and output according to the following requirements

#### Require:

1. **Writing the Literature Review:**
    - Read the provided abstracts and summarize their key contributions, methodologies, findings, and limitations.
    - Group related studies together based on themes, methods, or findings.
    - When citing a paper, use the format: `Author Year[Reference Number]` (e.g., Zhang 2019[1]).
    - Ensure the review is written in clear, professional, and academic language.

2. **Creating a Literature Matrix:**
    - After writing the literature review, extract key attributes of the studies (e.g., study goals, methods, datasets used, key findings, and limitations).
    - Summarize these attributes in a table format with columns for each attribute and rows for each paper.
    - Include a "Common Features" row to highlight similarities across the studies and a "Distinctive Features" row to outline differences.

#### Output Structure:

1. **Literature Review:**
    - A synthesized paragraph or set of paragraphs summarizing the studies.
    - Reply in en-GB.

2. **Literature Matrix:**
    - A structured table that compares and contrasts the studies.
    - Reply in en-GB.
```

**Comment by amateurProfessor** [comment]
### ❓AskZotero - English Version
Replies are in English, both English and German triggers (German original: FragZotero)
```
#❓AskZotero[color=#D53ECA][trigger=/^(frag zotero|zotero fragen|ähnliche artikel|ask zotero|query zotero|related articles)/]
You are a helpful assistant. Context informations are below from different papers.
${
Meet.Zotero.getLocalRelatedText(Meet.Global.input)
}$

Using the provided context information come from different paragraphs in different papers, write a comprehensive reply to the given question. Make sure to cite results using [number] notation after the reference. Use the first author and year to represent the article, for example, "Sun et al. (2020)". If the provided context information refer to multiple subjects with the same name, write separate answers for each subject. Use prior knowledge only if the given context didn't provide enough information.

Answer this question in an unbiased, comprehensive, and scholarly tone.
Reply in en-GB.

Question: ${Meet.Global.input}$

Answer:
```

**Comment by huachuman** [comment]
i would love to see this with google gemini or deepseek

**Reply by MuiseDestiny** [reply]
support
<img width="630" alt="image" src="https://github.com/user-attachments/assets/a89fe604-ff1a-4720-a8a4-e6d61d84f6c7" />
